server/src/main/java/org/opensearch/index/search/QueryStringQueryParser.java

```diff
@@ -384,6 +400,9 @@ protected Query getFieldQuery(String field, String queryText, int slop) throws P
         if (fields.isEmpty()) {
             return newUnmappedFieldQuery(field);
         }
+        for (Map.Entry<String, Float> entry : fields.entrySet()) {
```

Comment:
Not the right place: the `queryBuilder.parse` could return `null` (below), we probably should one add fields when the query is proven parseable

Reply:
done